var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -112,250 +104,192 @@ class Opencv(CMakePackage, CudaPackage):
 
-    # TODO For Cuda >= 10, make sure 'dynlink_nvcuvid.h' or 'nvcuvid.h'
-    # exists, otherwise build will fail
-    # See https://github.com/opencv/opencv_contrib/issues/1786
-    conflicts('^cuda@10:', when='+cudacodec')
     conflicts('+cuda', when='~contrib', msg='cuda support requires +contrib')
 
     # IPP is provided x86_64 only
     conflicts('+ipp', when="arch=aarch64:")
 
-    # Module opencv_calib3d disabled because opencv_flann dependency
-    # can't be resolved!
+    # OpenCV module conflicts
+    # These conflicts can be scraped from CMakeCache.txt, look for:
+    #     OPENCV_MODULE_opencv_*_REQ_DEPS
+    # If these required dependencies aren't found, CMake will silently
+    # disable the requested module
+    conflicts('+calib3d', when='~imgproc')
+    conflicts('+calib3d', when='~features2d')
     conflicts('+calib3d', when='~flann')
-
-    extends('python', when='+python')
+    conflicts('+dnn', when='~core')
+    conflicts('+dnn', when='~imgproc')
+    conflicts('+features2d', when='~imgproc')
+    conflicts('+flann', when='~core')
+    conflicts('+gapi', when='~imgproc')
+    conflicts('+highgui', when='~imgproc')
+    conflicts('+highgui', when='~imgcodecs')
+    conflicts('+imgcodecs', when='~imgproc')
+    conflicts('+imgproc', when='~core')
+    conflicts('+ml', when='~core')
+    conflicts('+objc_bindings_generator', when='~core')
+    conflicts('+objc_bindings_generator', when='~imgproc')
+    conflicts('+objdetect', when='~core')
+    conflicts('+objdetect', when='~imgproc')
+    conflicts('+objdetect', when='~calib3d')
+    conflicts('+photo', when='~imgproc')
+    conflicts('+python2', when='~python_bindings_generator')
+    conflicts('+python3', when='~python_bindings_generator')
+    conflicts('+stitching', when='~imgproc')
+    conflicts('+stitching', when='~features2d')
+    conflicts('+stitching', when='~calib3d')
+    conflicts('+stitching', when='~flann')
+    conflicts('+ts', when='~core')
+    conflicts('+ts', when='~imgproc')
+    conflicts('+ts', when='~imgcodecs')
+    conflicts('+ts', when='~videoio')
+    conflicts('+ts', when='~highgui')
+    conflicts('+video', when='~imgproc')
+    conflicts('+videoio', when='~imgproc')
+    conflicts('+videoio', when='~imgcodecs')
+    conflicts('+world', when='~core')
+
+    conflicts('+python2', when='+python3')
+    conflicts('+python3', when='+python2')
+
+    extends('python', when='+python2')
+    extends('python', when='+python3')
 
     def cmake_args(self):
         spec = self.spec
+        args = []
 
-        # Standard variants
-        args = [
-            '-DBUILD_SHARED_LIBS:BOOL={0}'.format((
-                'ON' if '+shared' in spec else 'OFF')),
-            '-DENABLE_PRECOMPILED_HEADERS:BOOL=OFF',
-            '-DWITH_LAPACK={0}'.format((
-                'ON' if '+lapack' in spec else 'OFF')),
-            '-DENABLE_POWERPC={0}'.format((
-                'ON' if '+powerpc' in spec else 'OFF')),
-            '-DENABLE_VSX={0}'.format((
```

Comment:
Did the VSX option get removed from the CMake package?

Reply:
It's not removed, but it is deprecated. I've never personally used it, but it's possible that it's important for folks on PowerPC. Should I add it back?